crates/sp-executor/src/lib.rs

```diff
@@ -133,6 +133,10 @@ impl<Hash: Encode, Number: Encode> From<ExecutionReceipt<Hash, Number>> for Opaq
 /// Fraud proof for the state computation.
 #[derive(Decode, Encode, TypeInfo, PartialEq, Eq, Clone, RuntimeDebug)]
 pub struct FraudProof {
+    /// State root before the fraudulent transaction.
+    pub pre_state_root: H256,
```

Comment:
Would't storage proof below include this automatically?

Reply:
Nope, the storage proof is only composed of the trie nodes touched during the execution.